model.py

```diff
@@ -1306,6 +1320,44 @@ def lookup_inside(self, name, using_overlap=False):
             )
         return places[0]
 
+    def lookup_one_through_external_source(self, name):
+        """Use an external source to find a Place that is a) inside `self`
+        and b) identifies the place human beings call `name`.
+
+        Currently the only way this might work is when using
+        uszipcodes to look up a city inside a state. In this case the result
+        will be a Place representing one of the city's postal codes.
+
+        :return: A Place, or None if the lookup fails.
+        """
+        if self.type != Place.STATE:
+            # uszipcodes keeps track of places in terms of their state.
+            return None
+
+        _db = Session.object_session(self)
+        search = uszipcode.SearchEngine(simple_zipcode=True)
```

Comment:
`simple_zipcode=True` means the data download (which happens once every couple of weeks) is a few megabytes instead of half a gigabyte.

Reply:
How do you track or implement that it downloads just once every couple of weeks?

Reply:
It's internal to `uszipcode`, I don't know for sure that that's the timeframe -- just going on what it says in the docs.